### PR TITLE
GH-981 make project creator CLI tool easily consumable

### PR DESCRIPTION
**Pull request by JanKoehnlein**
Build the scripts and bundle the libs in a zip.

### CODE REVIEW

File: org.eclipse.xtext.xtext.wizard/build.gradle [hunk 1]
```diff
@@ -3,11 +3,18 @@ description = 'Platform-independent component for creating new Xtext projects.'
 
 dependencies {
 	compile project(':org.eclipse.xtext')
-	optional "args4j:args4j:$versions.args4j"
+	compile "args4j:args4j:$versions.args4j"
```
Comment (cdietrich) [inline org.eclipse.xtext.xtext.wizard/build.gradle]: Am not sure if we want this

**Comment by kthoms** [comment]
This will produce the application, but how would it be distributed?

File: org.eclipse.xtext.xtext.wizard/build.gradle (hunk 1 above)
Comment (JanKoehnlein) [reply org.eclipse.xtext.xtext.wizard/build.gradle]: It won't build an executable application otherwise.

**Comment by JanKoehnlein** [comment]
Well, for me it'd be enough to download the ZIP from Jenkins. We could also push it to bintray or similar. What do you expect as distribution?

**Comment by kthoms** [comment]
It should go to Maven Central like everything else, likely with packaging zip. I don't think we should introduce another distribution kind. And the process should be automated. From Maven perspective this would be an attached artifact, not sure what to do for Gradle here to make it work. 
The resulting zip should be picked up by the [xtext-snapshots](https://ci.eclipse.org/xtext/job/xtext-snapshots/) build job and be deployed.

**Comment by kthoms** [comment]
> Well, for me it'd be enough to download the ZIP from Jenkins. 
This would be an undefined state and not usable from clients that would like to use a release version from a scripted build (e.g. to create a Yeoman plugin).

**Comment by kthoms** [comment]
Could be something that could be configured for the Gradle [application plugin](https://docs.gradle.org/current/userguide/application_plugin.html), but I can't read it from the docs how to do that.

**Comment by cdietrich** [comment]
dependency is now non optional on master. i dont think we should publish an additional zip to somewhere. i dont even know if the typefox publishing plugin could handle that

**Pull request closed without merging**